src/v/cluster/topic_table.cc

```diff
@@ -277,6 +277,36 @@ topic_table::apply(update_topic_properties_cmd cmd, model::offset o) {
     co_return make_error_code(errc::success);
 }
 
+ss::future<std::error_code>
+topic_table::apply(create_materialized_topic_cmd cmd, model::offset o) {
+    const model::topic_namespace& source = cmd.value;
+    const model::topic_namespace& new_materialized_topic = cmd.key;
+    if (_topics.contains(new_materialized_topic)) {
+        co_return make_error_code(errc::topic_already_exists);
+    }
+    auto tp = _topics.find(source);
+    if (tp == _topics.end()) {
+        /// Source topic must exist if attempting to create a materialized topic
+        co_return make_error_code(errc::topic_invalid_config);
+    }
+
+    for (const auto& pas : tp->second.configuration.assignments) {
+        _pending_deltas.emplace_back(
+          model::ntp(
+            new_materialized_topic.ns, new_materialized_topic.tp, pas.id),
+          pas,
+          o,
+          delta::op_type::add_materialized);
+    }
+
+    topic_table::topic_metadata tca(tp->second);
+    tca.configuration.cfg.tp_ns.tp = new_materialized_topic.tp;
+    tca.configuration.cfg.properties.source_topic = source.tp();
```

Comment:
what about cleanup policy, segment size, retention settings ? should it be inherited from source topic ?

Reply:
`tca` is copied and only the interested portions are overwritten. So therefore all settings of source topic are inherited.